userprefs.php

```diff
@@ -951,54 +945,18 @@ function _show_credit_name_adhoc()
 {
     global $userSettings;
 
-    $credit_names = ['real_name', 'username', 'other'];
-    $credit_names_labels = [_('Real Name'), _('Username'), _('Other') . ":"];
-    $credit_name_value = $userSettings->get_value('credit_name', 'real_name');
-    $on_change = "f.credit_other.disabled = (t.options[t.selectedIndex].value!='other');";
-    dropdown_select_values_and_labels('credit_name', $credit_name_value, $credit_names, $credit_names_labels, $on_change);
-    echo " ";
-
-    $credit_other_value = attr_safe($userSettings->get_value('credit_other', ''));
-    echo "<input type='text' name='credit_other' value='$credit_other_value'>\n";
-}
-
-// The third argument should be a 'real' array.
-// The labels will be displayed to the user,
-// one of the values will be passed back from the browser as the selected value.
-//
-// The fifth (optional argument), $on_change, is used as a javascript event handler
-// on the dropdown. It will be made into a function so quote marks should not
-// be any problems.
-// Example value: "alert('Hi'+\"!\");"
-// Using this as the $on_change-argument will popup an alert displaying the string
-// 'Hi!' (without quotes).
-// The use of these event handlers are foremost to enable/disable certain preferences
-// depending on the values set in other preferences.
-//
-// The event handler will also be run on page-load and in order to achieve this,
-// something resembling a hack has been introduced. Always refer to the form
-// as the variable f, and always use the variable t to refer to the dropdown.
-// DO NOT USE this.form and this, respectively!!!
-function dropdown_select_values_and_labels($field_name, $current_value, $values, $labels, $on_change = '')
-{
-    global $event_id, $window_onload_event;
-
-    $function_name = 'event' . ++$event_id;
-    $jscode = "var f=document.forms[0];\nvar t=f.$field_name;\n$on_change";
+    $credit_options = [
+        'real_name' => _('Real Name'),
+        'username' => _('Username'),
+        'other' => _('Other:'),
```

Comment:
```suggestion
        'other' => _('Other') . ':',
```
We don't want the `:` inside the translated string.